instant-xml-macros/src/de.rs

```diff
@@ -0,0 +1,264 @@
+use proc_macro2::{Ident, Span, TokenStream};
+use quote::quote;
+
+use crate::{get_namespaces, retrieve_attr};
+
+struct Tokens<'a> {
+    enum_: &'a mut TokenStream,
+    consts: &'a mut TokenStream,
+    names: &'a mut TokenStream,
+    match_: &'a mut TokenStream,
+}
+
+pub struct Deserializer {
+    fn_vec: Vec<TokenStream>,
+}
+
+impl quote::ToTokens for Deserializer {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        let vec = &self.fn_vec;
+        tokens.extend(quote!(
+            #(#vec)*
+        ));
+    }
+}
+
+impl Deserializer {
+    pub fn new(input: &syn::DeriveInput) -> Deserializer {
+        let ident = &input.ident;
+        let name = ident.to_string();
+        let mut fn_vec = Vec::new();
```

Comment:
It looks to me like `fn_vec` is just the output token stream. Maybe it should be called `out` or something and be of type `TokenStream`? You can append a `TokenStream` to another by using `TokenStream::extend()` IIRC.

Reply:
done